File: MERN/Week 1/Day 2/Practice/Ninja/ninja.js
class Ninja {
    constructor (name ){
        this.name = name
        this.health = 90 ;
        this.speed = 3 ;
        this.strength = 3 ;
    }
    sayName(){
        console.log(this.name)
    }

    showStats(){
        console.log( `  ${this.name} my health is ${this.health} ,  my speed is ${this.speed} and my strength is ${this.strength}` )
    }
    
    drinkShake(){
        this.health +=10 ;
        console.log(this.health)


    }
}

const ninja1 = new Ninja("Hyabusa");
ninja1.sayName();
ninja1.showStats();
ninja1.drinkShake();

File: Python/Week 1/Day 2/Optional/ninjas_vs_pirates/classes/ninja.py
from classes.character import Character

class Ninja(Character):

    def __init__( self , name ):
        super().__init__(name=name,strength=10,speed=5,health=100)
        self.stealth = self.strength * 2

    def perform_stealth_attack(self,target) :
        print(f"{self.name} the ninja perfomrs a stealth attack§")   
        
    

    def attack( self , pirate ):
        pirate.health -= self.strength
        return self